contribs/emissions/src/main/java/org/matsim/contrib/emissions/WarmEmissionHandler.java

```diff
@@ -174,12 +191,12 @@ private void emissionsCalculation(Id<Vehicle> vehicleId, Vehicle vehicle, Link l
 	private boolean warnIfZeroLinkLength( Id<Link> linkId, double linkLength) {
 		if (linkLength == 0.) {
 			if (zeroLinkLengthWarnCnt == 0) {
-				logger.warn("Length of the link " + linkId + " is zero. No emissions will be estimated for this link. Make sure, this is intentional.");
+				logger.warn("Length of the link " + linkId + " is zero. No emissions will be estimated for this link. Make sure that this is intentional.");
 				logger.warn(Gbl.ONLYONCE);
 				zeroLinkLengthWarnCnt++;
 			}
-			return true;
-		} else { return false; }
+			return false;
+		} else { return true; }
```

Comment:
Why have you changed the return values?
I would expect the previous ones: If linkLength == 0 -> true, if not -> false.